File: ruby/rbsearch/searchfile.rb
################################################################################
#
# searchfile.rb
#
# class SearchFile: encapsulates a file to be searched
#
################################################################################

class SearchFile

  attr_accessor :containers
  attr_accessor :path
  attr_accessor :filename
  attr_accessor :filetype

  @CONTAINER_SEPARATOR = '!'

  def initialize(path, filename, filetype)
    @containers = []
    @path = path
    @filename = filename
    @filetype = filetype
  end

  def to_s
    s = ""
    if @containers.length > 0
      @containers.join(@CONTAINER_SEPARATOR) + @CONTAINER_SEPARATOR
    end
    s += Pathname.new(@path).join(@filename).to_s
    s
  end

end

File: typescript/tssearch/src/searchfile.ts
/*
 * searchfile.js
 *
 * encapsulates a file to be searched
 */

"use strict";

import {FileType} from './filetype';

var path = require('path');

export class SearchFile {
    containerSeparator: string = '!';
    containers: string[] = [];
    pathname: string;
    filename: string;
    filetype: FileType;

    constructor(pathname: string, filename: string, filetype: FileType) {
        this.pathname = pathname;
        this.filename = filename;
        this.filetype = filetype;
    }

    public relativePath(): string {
        if (this.pathname === '.' || this.pathname === './') return './' + this.filename;
        return path.join(this.pathname, this.filename);
    };

    public toString(): string {
        let s: string = '';
        if (this.containers.length > 0) {
            s = this.containers.join(this.containerSeparator) + this.containerSeparator;
        }
        s += path.join(this.pathname, this.filename);
        return s;
    }
}
